src/agent/adu_core_interface/tests/adu_core_interface_ut.cpp

```diff
@@ -70,21 +71,21 @@ class ADUC_UT_ReportPropertyAsyncValues
     void
     set(ADUC_ClientHandle deviceHandleIn,
         const unsigned char* reportedStateIn,
-        size_t reportedStateLenIn,
+        size_t _reportedStateLenIn,
         IOTHUB_CLIENT_REPORTED_STATE_CALLBACK reportedStateCallbackIn,
         void* userContextCallbackIn)
     {
         deviceHandle = deviceHandleIn;
 
-        memcpy(reportedState, reportedStateIn, reportedStateLenIn);
+        reportedState = reinterpret_cast<const char*>(reportedStateIn);
```

Comment:
>reportedState [](http://example.com/codeflow?start=8&length=13)

not safe.  there's a length passed which implies to me that this might not be null terminated.

if you want to convert to std::string then you should use the ctor that specifies a size, I **think** it's this one:

`from sequence (5) 
string (const char* s, size_t n);`

Reply:
good point